python/hsfs/feature_group.py

```diff
@@ -1675,6 +1698,151 @@ def insert(
+            object that implements the same `multi_part_insert` API.
+            ```python
+            feature_group = fs.get_or_create_feature_group("fg_name", version=1)
+
+            with feature_group.multi_part_insert() as writer:
+                # run inserts in a loop:
+                while loop:
+                    small_batch_df = ...
+                    writer.multi_part_insert(small_batch_df)
+            ```
+            The writer batches the small Dataframes and transmits them to Hopsworks
+            efficiently.
+            When exiting the context, the feature group writer is sure to exit
+            only once all the rows have been transmitted.
+
+        !!! example "Multi part insert with manual context management"
+            Instead of letting Python handle the entering and exiting of the
+            multi part insert context, you can start and finalize the context
+            manually.
+            ```python
+            feature_group = fs.get_or_create_feature_group("fg_name", version=1)
+
+            while loop:
+                small_batch_df = ...
+                feature_group.multi_part_insert(small_batch_df)
+
+            # IMPORTANT: finalize the multi part insert to make sure all rows
+            # have been transmitted
+            feature_group.finalize_multi_part_insert()
+            ```
+            Note that the first call to `multi_part_insert` initiates the context
+            and be sure to finalize it. The `finalize_multi_part_insert` is a
+            blocking call that returns once all rows have been transmitted.
+
+        # Arguments
+            features: DataFrame, RDD, Ndarray, list. Features to be saved.
+            overwrite: Drop all data in the feature group before
+                inserting new data. This does not affect metadata, defaults to False.
+            operation: Apache Hudi operation type `"insert"` or `"upsert"`.
+                Defaults to `"upsert"`.
+            storage: Overwrite default behaviour, write to offline
+                storage only with `"offline"` or online only with `"online"`, defaults
+                to `None`.
+            write_options: Additional write options as key-value pairs, defaults to `{}`.
+                When using the `python` engine, write_options can contain the
+                following entries:
+                * key `spark` and value an object of type
+                [hsfs.core.job_configuration.JobConfiguration](../job_configuration)
+                  to configure the Hopsworks Job used to write data into the
+                  feature group.
+                * key `wait_for_job` and value `True` or `False` to configure
+                  whether or not to the insert call should return only
+                  after the Hopsworks Job has finished. By default it waits.
+                * key `start_offline_backfill` and value `True` or `False` to configure
```

Comment:
`start_offline_backfill` is False by default in multi part insert.

Reply:
changed, plus added comment above on how to start backfill job once the multi part insert is done.